proxy/http/Http1ClientTransaction.cc

```diff
@@ -63,11 +63,9 @@ Http1ClientTransaction::set_parent(ProxyClientSession *new_parent)
 void
 Http1ClientTransaction::transaction_done()
 {
-  current_reader = nullptr;
   // If the parent session is not in the closed state, the destroy will not occur.
   if (parent) {
-    parent->destroy();
-    parent = nullptr;
+    dynamic_cast<Http1ClientSession *>(parent)->release_transaction();
```

Comment:
Yes, if you're not checking the result you might as well use `static_cast` which is  faster.